pkg/controller/clusterconnection/clusterconnection_controller.go

```diff
@@ -254,88 +244,85 @@ func (r *ReconcileConnection) Reconcile(ctx context.Context, request reconcile.R
-	// requirements to reconcile NetworkPolicy have been met.
-	//
-	// * In managed clusters, the License can only be pushed once Guardian has been deployed, and (as always) the Tier
-	//   can only be created once the License is available.
-	includeNetworkPolicy := true
-	var egressAccessControlFeatureRequired bool
-	var networkPolicyIsReconcilable bool
-	if clusterAddrHasDomain, err := managementClusterAddrHasDomain(managementClusterConnection); err == nil && clusterAddrHasDomain {
-		egressAccessControlFeatureRequired = true
-		networkPolicyIsReconcilable = utils.IsV3NetworkPolicyReconcilable(ctx, r.Client, networkpolicy.TigeraComponentTierName, common.EgressAccessControlFeature)
+	// Ensure the allow-tigera tier exists, before rendering any network policies within it.
+	includeV3NetworkPolicy := false
+	egressAccessControlFeatureRequired := false
+	if err := r.Client.Get(ctx, client.ObjectKey{Name: networkpolicy.TigeraComponentTierName}, &v3.Tier{}); err != nil {
+		// The creation of the Tier depends on this controller to reconcile it's non-NetworkPolicy resources so that the
+		// License becomes available. Therefore, if we fail to query the Tier, we exclude NetworkPolicy from reconciliation
+		// and tolerate errors arising from the Tier not being created.
+		if !k8serrors.IsNotFound(err) {
+			log.Error(err, "Error querying allow-tigera tier")
+			r.status.SetDegraded("Error querying allow-tigera tier", err.Error())
+			return reconcile.Result{}, err
+		}
 	} else {
-		if err != nil {
+		includeV3NetworkPolicy = true
+
+		if clusterAddrHasDomain, err := managementClusterAddrHasDomain(managementClusterConnection); err == nil && clusterAddrHasDomain {
+			egressAccessControlFeatureRequired = true
+		} else if err != nil {
 			log.Error(err, fmt.Sprintf(
 				"Failed to parse ManagementClusterAddr. Assuming %s does not require license feature %s",
 				render.GuardianPolicyName,
 				common.EgressAccessControlFeature,
 			))
 		}
-		egressAccessControlFeatureRequired = false
-		networkPolicyIsReconcilable = utils.IsV3NetworkPolicyReconcilable(ctx, r.Client, networkpolicy.TigeraComponentTierName)
 	}
 
-	if networkPolicyIsReconcilable {
-		if egressAccessControlFeatureRequired {
-			license, err := utils.FetchLicenseKey(ctx, r.Client)
-			if err != nil {
-				if k8serrors.IsNotFound(err) {
-					r.status.SetDegraded("License not found", err.Error())
-					return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
-				}
-				r.status.SetDegraded("Error querying license", err.Error())
-				return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
-			}
-
-			if !utils.IsFeatureActive(license, common.EgressAccessControlFeature) {
-				r.status.SetDegraded("Feature is not active", "License does not support feature: egress-access-control")
+	if egressAccessControlFeatureRequired {
+		license, err := utils.FetchLicenseKey(ctx, r.Client)
```

Comment:
The license check was not present prior to the allow-tigera work and was added for the feature check, so I was thinking it would be best to limit the scope of this check. We also need to be careful with the timing of a license check, as this controller is initially a dependency for the license being deployed

I've reorganized to make these considerations a bit more clear, but the logic should be equivalent